makefiles/binutils.mk

```diff
@@ -0,0 +1,116 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUBPROJECTS      += binutils
+BINUTILS_VERSION := 2.36.1
+DEB_BINUTILS_V   ?= $(BINUTILS_VERSION)
+
+BINUTILS_TARGETS := aarch64-linux-gnu aarch64-linux-musl alpha-linux-gnu alpha-linux-musl arm-linux-gnueabi arm-linux-gnueabihf arm-linux-musl hppa-linux-gnu hppa-linux-musl i686-kfreebsd-gnu i686-linux-gnu i686-linux-musl ia64-linux-gnu ia64-linux-musl m68k-linux-gnu m68k-linux-musl mips64el-linux-gnuabi64 mips64el-linux-musl mipsel-linux-gnu mipsel-linux-musl powerpc-linux-gnu powerpc-linux-musl powerpc64-linux-gnu powerpc64-linux-musl powerpc64le-linux-gnu powerpc64le-linux-musl riscv64-linux-gnu riscv64-linux-musl s390x-linux-gnu s390x-linux-musl sh4-linux-gnu sh4-linux-musl sparc64-linux-gnu sparc64-linux-musl x86_64-linux-gnu x86_64-linux-musl powerpc-apple-darwin x86_64-apple-darwin i686-apple-darwin arm-apple-darwin aarch64-apple-darwin i686-hurd-gnu i686-elf x86_64-elf arm-none-eabi aarch64-elf riscv64-elf alpha-elf ia64-elf m68k-elf powerpc-elf powerpc64-elf powerpc64le-elf mipsel-elf mipsel64-elf hppa-elf s390x-elf sh4-elf sparc64-elf i686-unknown-haiku x86_64-unknown-haiku arm-unknown-haiku aarch64-unknown-haiku
+
+BINUTILS_CONFARGS := --enable-obsolete \
+	--enable-shared \
+	--enable-plugins \
+	--enable-threads \
+	--with-system-zlib \
+	--enable-deterministic-archives \
+	--disable-compressed-debug-sections \
+	--enable-new-dtags \
+	--disable-x86-used-note \
+	--with-pkgversion="GNU Binutils for Procursus" \
+	--enable-ld=default \
+	--enable-gold \
+	--enable-default-hash-style=gnu
+
+binutils-setup: setup
+	wget -q -nc -P $(BUILD_SOURCE) https://ftpmirror.gnu.org/binutils/binutils-$(BINUTILS_VERSION).tar.xz{,.sig}
+	$(call PGP_VERIFY,binutils-$(BINUTILS_VERSION).tar.xz)
+	$(call EXTRACT_TAR,binutils-$(BINUTILS_VERSION).tar.xz,binutils-$(BINUTILS_VERSION),binutils)
+	$(call DO_PATCH,binutils,binutils,-p1)
+
+ifneq ($(wildcard $(BUILD_WORK)/binutils/.build_complete),)
+binutils:
+	@echo "Using previously built binutils."
+else
+binutils: binutils-setup $(patsubst %,%-binutils-target,$(BINUTILS_TARGETS))
+	touch $(BUILD_WORK)/binutils/.build_complete
+endif
+
+%-binutils-target: binutils-setup gettext
+	target=$$(echo $@ | rev | cut -d- -f3- | rev); \
+	if [ "$$GNU_HOST_TRIPLE" = "$$target" ]; then \
+		binutils_extra_flags=--program-prefix=g; \
+	fi; \
+	if [ -f $(BUILD_WORK)/binutils/$$target/.build_complete ]; then \
+		echo "Using previously built $$target binutils"; \
+	else \
+		mkdir -p $(BUILD_WORK)/binutils/$$target; \
+		cd $(BUILD_WORK)/binutils/$$target && ../configure -C \
+			--build=$$($(BUILD_MISC)/config.guess) \
+			--host=$(GNU_HOST_TRIPLE) \
+			--prefix=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX) \
+			--libdir=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/$(GNU_HOST_TRIPLE)/$$target/lib \
+			--includedir=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/$(GNU_HOST_TRIPLE)/$$target/include \
```

Comment:
You may want to do `MEMO_TARGET` here not `GNU_HOST_TRIPLE`.

Reply:
thinking about `LLVM_TARGET`

Reply:
However, `LLVM_TARGET` has the min version in it, which we don't want.

Reply:
How about `RUST_TARGET`?
(I know this is not rust)

Reply:
However, `RUST_TARGET` is shared between audioOS and TvOS, and iOS arm64 and iOS arm64e. `MEMO_TARGET` is the only one that isn't used for multiple targets.